docs/common-sdk-requirements.md

```diff
@@ -0,0 +1,43 @@
+# Requirements for the Common SDK infrastructure
+The purpose of this doc is to have a place to list and discuss all requirements to the new modular SDK infrastructure (also informally called "the Common SDK infrastructure") before creating an official design.
+
+The information listed here was primarily sourced in a few meetings/discussions/whiteboardings we had before.
+
+## Terminology
+* _Common SDK_ is the new modular SDK. 
+* _Service_ is a single building unit (module) of the new modular SDK. TODO: please let me know if another term is desired for this.
+
+## Requirements
+### Requirements to the SDK
+* Greater degree of flexibility and scalability in development, maintenance, customization and extensibility due to:
+    * Non-monolithic, modular structure
+    * Smaller building blocks (services) that can be worked on independently
+    * High configurability (configuration-driven design)
+* Configurable composition:
+    * Services can be selected
+    * Services can be optional
+    * Services can be required
+    * Composition rules/restrictions apply based on each service dependencies and configuration
+    * Configuration errors must be prevented/communicated at the time of configuring
+* Ability to tailor SDK to specific customer needs
+* Can be composed and configured by a customer
+* Trackable SDK generation and download process to provide acceptable customer experience
+* Can be used to support all existing SDK platforms
+* Can be integrated into existing SDK generation process (eventually replace generation of monolithic SDKs)
+* Support of built-in (authored by PlayFab) and custom services (created by customers)
+* Service management/administration system (for PlayFab admins)
+* Well-defined and easy new service development process
+* Well-defined and easy new service registration process
+
+### Requirements to a service
```

Comment:
Should  we have a registry for plugin discoverability/ plugin registry ?

Reply:
Absolutely. I think Nuget is the model to use here. Hell, C# and C++ could piggyback DIRECTLY on Nuget...